pkg/scanners/trivy/trivy.go

```diff
@@ -241,22 +242,33 @@ func initScanner(ctx context.Context) (Scanner, error) {
 func scan(s Scanner, allImages []eraserv1alpha1.Image) ([]eraserv1alpha1.Image, []eraserv1alpha1.Image) {
 	vulnerableImages := make([]eraserv1alpha1.Image, 0, len(allImages))
 	failedImages := make([]eraserv1alpha1.Image, 0, len(allImages))
-
-	for _, img := range allImages {
-		// Logs scan failures
-		status, err := s.Scan(img)
-		if err != nil {
-			failedImages = append(failedImages, img)
-			log.Error(err, "scan failed")
-			continue
-		}
-
-		switch status {
-		case StatusNonCompliant:
-			log.Info("vulnerable image found", "img", img)
-			vulnerableImages = append(vulnerableImages, img)
-		case StatusFailed:
-			failedImages = append(failedImages, img)
+	// track total scan job time
+	timer := time.NewTimer(*imageScanTotalTimeout)
+
+	for idx, img := range allImages {
+		select {
+		case <-timer.C:
+			log.Info("image scan total timeout exceeded", "timeout", imageScanTotalTimeout.String())
+			if idx != len(allImages) {
+				failedImages = append(failedImages, allImages[idx+1:]...)
```

Comment:
Why `idx+1`? If we reach the timeout before scanning the image with index `idx`, we should include that image in the failed images.

Since `scan` is a private helper function, it's fine to change the signature to return an error.